pkg/common/vclib/datacenter.go

```diff
@@ -383,6 +396,135 @@ func (dc *Datacenter) GetAllDatastoreClusters(ctx context.Context, child bool) (
 	return spURLInfoMap, nil
 }
 
+// GetDatastoreClusterByName gets the DatastoreCluster object for the given name
+func (dc *Datacenter) GetDatastoreClusterByName(ctx context.Context, name string) (*StoragePodInfo, error) {
+	finder := getFinder(dc)
+	ds, err := finder.DatastoreCluster(ctx, name)
+	if err != nil {
+		glog.Errorf("Failed while searching for datastore cluster: %s. err: %+v", name, err)
+		return nil, err
+	}
+
+	var spMo mo.StoragePod
+	pc := property.DefaultCollector(dc.Client())
+	properties := []string{StoragePodDrsEntryProperty, StoragePodProperty}
+	err = pc.RetrieveOne(ctx, ds.Reference(), properties, &spMo)
+	if err != nil {
+		glog.Errorf("Failed to get Datastore managed objects from datastore objects."+
+			" properties: %+v, err: %v", properties, err)
+		return nil, err
+	}
+
+	return &StoragePodInfo{
+		&StoragePod{
+			dc,
+			object.NewStoragePod(dc.Client(), spMo.Reference()),
+			make([]*Datastore, 0),
+		},
+		spMo.Summary,
+		&spMo.PodStorageDrsEntry.StorageDrsConfig,
+		make([]*DatastoreInfo, 0),
+	}, nil
+}
+
+func (dc *Datacenter) CreateFirstClassDisk(ctx context.Context,
+	datastoreName string, datastoreType ParentDatastoreType,
+	diskName string, diskSize int64) error {
+
+	m := vslm.NewObjectManager(dc.Client())
+
+	var pool *object.ResourcePool
+	var ds types.ManagedObjectReference
+	if datastoreType == TypeDatastoreCluster {
+		storagePod, err := dc.GetDatastoreClusterByName(ctx, datastoreName)
+		if err != nil {
+			glog.Errorf("GetDatastoreClusterByName failed. Err: %v", err)
+			return err
+		}
+		ds = storagePod.Reference()
+
+		pool, err = dc.GetResourcePool(ctx, "")
+		if err != nil {
+			glog.Errorf("GetResourcePool failed. Err: %v", err)
+			return err
+		}
+	} else {
+		datastore, err := dc.GetDatastoreByName(ctx, datastoreName)
+		if err != nil {
+			glog.Errorf("GetDatastoreByName failed. Err: %v", err)
+			return err
+		}
+		ds = datastore.Reference()
+	}
+
+	spec := types.VslmCreateSpec{
+		Name:         diskName,
+		CapacityInMB: diskSize,
+		BackingSpec: &types.VslmCreateSpecDiskFileBackingSpec{
+			VslmCreateSpecBackingSpec: types.VslmCreateSpecBackingSpec{
+				Datastore: ds,
+			},
+			ProvisioningType: string(types.BaseConfigInfoDiskFileBackingInfoProvisioningTypeThin),
+		},
+	}
+
+	if datastoreType == TypeDatastoreCluster {
+		err := m.PlaceDisk(ctx, &spec, pool.Reference())
+		if err != nil {
+			glog.Errorf("PlaceDisk failed. Err: %v", err)
+			return err
+		}
+	}
+
+	task, err := m.CreateDisk(ctx, spec)
+	if err != nil {
+		glog.Errorf("CreateDisk failed. Err: %v", err)
+		return err
+	}
+
+	err = task.Wait(ctx)
+	if err != nil {
+		glog.Errorf("Wait failed. Err: %v", err)
```

Comment:
More context here would be good, `"CreateDisk(%s) failed. Err: %v", diskName, err)`

Reply:
Agreed. Will update.